### PR TITLE
Fixed ReCaptcha position in user registration form

### PR DESCRIPTION
**Pull request by shazmasiddiqui**
Pull Request for Issue #37698

### Testing Instructions
Enable ReCaptcha and custom fields in registration form and check the position of ReCaptcha. It should be placed at the bottom of the form, right before Register button.

### Actual result BEFORE applying this Pull Request
The ReCaptcha is placed before custom fields, in the middle of the form

### Expected result AFTER applying this Pull Request
The ReCaptcha should be placed before the Register button

### CODE REVIEW

File: components/com_users/tmpl/registration/default.php [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 	<form id="member-registration" action="<?php echo Route::_('index.php?option=com_users&task=registration.register'); ?>" method="post" class="com-users-registration__form form-validate" enctype="multipart/form-data">
 		<?php // Iterate through the form fieldsets and display each one. ?>
+		<?php $captcha=''; ?>
```
Comment (brianteeman) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
		<?php $captcha = ''; ?>
```

File: components/com_users/tmpl/registration/default.php [hunk 2]
```diff
@@ -34,10 +35,19 @@
 					<?php if (isset($fieldset->label)) : ?>
 						<legend><?php echo Text::_($fieldset->label); ?></legend>
 					<?php endif; ?>
-					<?php echo $this->form->renderFieldset($fieldset->name); ?>
+					<?php foreach ($fields as $field): ?>
```
Comment (brianteeman) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
					<?php foreach ($fields as $field): ?>
```

remove tabs at end of line

File: components/com_users/tmpl/registration/default.php [hunk 3]
```diff
@@ -34,10 +35,19 @@
 					<?php if (isset($fieldset->label)) : ?>
 						<legend><?php echo Text::_($fieldset->label); ?></legend>
 					<?php endif; ?>
-					<?php echo $this->form->renderFieldset($fieldset->name); ?>
+					<?php foreach ($fields as $field): ?>			
+						<?php if ($field->name=="jform[captcha]"): ?>
```
Comment (brianteeman) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
						<?php if ($field->name=="jform[captcha]"): ?>
```

remove space at end of line

File: components/com_users/tmpl/registration/default.php [hunk 4]
```diff
@@ -34,10 +35,19 @@
...
+						<?php else: ?>
+							<?php echo $field->renderField();?>
+						<?php endif; ?>		
+					<?php endforeach; ?>
 				</fieldset>
 			<?php endif; ?>
 		<?php endforeach; ?>
+		<?php if($captcha): ?>
+			<?php echo $captcha->renderField();?>
+		<?php endif; ?>
```
Comment (brianteeman) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
		<?php endif; ?>
```

remove tab at end of line

File: components/com_users/tmpl/registration/default.php [hunk 5]
```diff
@@ -34,10 +35,19 @@
 					<?php if (isset($fieldset->label)) : ?>
 						<legend><?php echo Text::_($fieldset->label); ?></legend>
 					<?php endif; ?>
-					<?php echo $this->form->renderFieldset($fieldset->name); ?>
+					<?php foreach ($fields as $field): ?>
+						<?php if ($field->name=="jform[captcha]"): ?>
+							<?php $captcha=$field; ?>
```
Comment (richard67) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
						<?php if ($field->name === 'jform[captcha]'): ?>
							<?php $captcha = $field; ?>
```
Please use type safe comparison and correct code style (one space before and after assignment or comparison operators, single quotes for string)

File: components/com_users/tmpl/registration/default.php [hunk 6]
```diff
@@ -34,10 +35,19 @@
...
+						<?php else: ?>
+							<?php echo $field->renderField();?>
+						<?php endif; ?>		
+					<?php endforeach; ?>
 				</fieldset>
 			<?php endif; ?>
 		<?php endforeach; ?>
+		<?php if($captcha): ?>
```
Comment (richard67) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
		<?php if ($captcha): ?>
```
Code style: One space between if and the opening bracket for the condition.

**Comment by richard67** [comment]
@shazmasiddiqui Check our coding standards: https://developer.joomla.org/coding-standards/introduction.html . And a useful hint: Adjust your editor in a way so that you can see and distinguish tabs and spaces.

**Comment by shazmasiddiqui** [comment]
@richard67 Thankyou for your helpful suggestions. I will keep these suggestions in mind for my next PR's.

**Comment by richard67** [comment]
@shazmasiddiqui You don't always need to update your branch when it is shown as not up to date with the base branch on GitHub. You only need to do that when GitHub shows that there are conflicting files, which was not the case.

File: components/com_users/tmpl/registration/default.php [hunk 7]
```diff
@@ -34,10 +35,19 @@
...
+						<?php else: ?>
+							<?php echo $field->renderField();?>
+						<?php endif; ?>		
+					<?php endforeach; ?>
 				</fieldset>
 			<?php endif; ?>
 		<?php endforeach; ?>
+		<?php if ($captcha): ?>
+			<?php echo $captcha->renderField();?>
+		<?php endif; ?>
```
Comment (Quy) [inline components/com_users/tmpl/registration/default.php]:
```suggestion
					<?php foreach ($fields as $field) : ?>
						<?php if ($field->name === 'jform[captcha]') : ?>
							<?php $captcha = $field; ?>
						<?php else: ?>
							<?php echo $field->renderField(); ?>
						<?php endif; ?>		
					<?php endforeach; ?>
				</fieldset>
			<?php endif; ?>
		<?php endforeach; ?>
		<?php if ($captcha) : ?>
			<?php echo $captcha->renderField(); ?>
		<?php endif; ?>
```

**Comment by HLeithner** [comment]
This pull requests has automatically rebased to 4.2-dev.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.3-dev.

**Comment by obuisard** [comment]
Shazma @shazmasiddiqui, does the PR #38604 fix the issue?

**Comment by obuisard** [comment]
This has been addressed with https://github.com/joomla/joomla-cms/pull/38604.
Thank you for your PR!

**Pull request closed without merging**